graph_generation/generate_from_config.py

```diff
@@ -0,0 +1,230 @@
+"""
+This module generates a random data-dependancy mapping graph from config file.
+You can create graph in proto format, or in networkx format.
+
+Usage:
+    python3 graph_generation/generate_from_config.py \
+         --output_file "output.graphml" \
+         --config_file "graph_generation/configs/config_15_09_20.yaml" \
+         --graph_type "networkx" \
+         --overwrite
+"""
+
+import yaml
+import time
+import logging
+import argparse
+
+from connection_generator import ConnectionGenerator
+from attribute_generator import AttributeGenerator
+from graph_template import GraphTemplate
+
+from config_params.collection_params import CollectionParams
+from config_params.data_integrity_params import DataIntegrityParams
+from config_params.dataset_params import DatasetParams
+from config_params.dataset_to_system_params import DatasetToSystemParams
+from config_params.processing_params import ProcessingParams
+from config_params.system_params import SystemParams
+from config_params.connection_params import ConnectionParams
+
+
+def process_map(config_map, proba=False, enum=False):
+    """Converts string map in format [key1:value1 key2:value2] into a dictionary.
+       key - represents a count of connections (integer) or enums (string).
+       value - represents a count of occurrences (integer) or probability (float).
+   Args:
+       config_map: string that contains the map
+       proba: boolean, true if values are probabilities, not counts.
+       enum: boolean, true if keys are enums, not integers.
+
+   Returns:
+       A dictionary of keys and values in converted format.
+   """
+    processed_map = {j[0]: j[1] for j in [i.split(":") for i in config_map.strip("[]").split()]}
+    keys = [int(k) if not enum else k for k in processed_map.keys()]
+    values = [int(v) if not proba else float(v) for v in processed_map.values()]
+    return {keys[i]: values[i] for i in range(len(keys))}
+
+
+def parse_args():
+    """Parses input arguments."""
+    parser = argparse.ArgumentParser(description='Generate graph based on config.')
+    parser.add_argument('-f', '--output_file', help='Path to output graphml or proto binary.', required=True)
+    parser.add_argument('-c', '--config_file', help='Path to yaml config file with graph parameters.', required=True)
+    parser.add_argument('-t', '--graph_type', help='Type of the graph to generate. Can be proto or networkx.',
+                        default="networkx", choices=["proto", "networkx"])
+    parser.add_argument('-o', '--overwrite', help='If output file exists, overwrite it.', type=bool, default=False)
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == '__main__':
```

Comment:
The main method is quite long (150+ lines). It's good practice to have a few lines only in the main (move the logic to another function), and have short functions (<100 lines) when possible